robottelo/host_helpers/cli_factory.py

```diff
@@ -638,7 +638,7 @@ def setup_org_for_a_custom_repo(self, options=None):
         lce_promoted = cv_info['lifecycle-environments']
         # Promote version to next env
         try:
-            if env_id not in [int(lce['id']) for lce in lce_promoted]:
+            if (env_id not in [int(lce['id']) for lce in lce_promoted]) or multi_repo:
```

Comment:
Explain in `#` comment please

Reply:
It's not clear, what is multi_repo arg here could you elaborate? and when multi_repo is True, then shouldn't we check  that first then the other check?

Reply:
Can someone from the Phoenix-content group chime in and provide feedback on it?
@vsedmik @ColeHiggins2 @sambible @vijaysawant

Reply:
Eh, it seems that @Griffin-Sullivan is trying to bypass @vijaysawant's bypass.

Looking at [the first bypass PR](https://github.com/SatelliteQE/robottelo/pull/12309), my guess is that it was trying to resolve the CVV Sorting Issue, which was later fixed on line 638.

Anyway, reading the docstring I don't think this helper was meant to be called in a loop, not for adding repos. It was for wrapping a repo into a product and shipping it via CV and AK.
If we need to create more custom repos in one custom product, we could extend the helper and pass the repo urls as a list and add them inside the helper.

Also, wouldn't it be easier for you @Griffin-Sullivan use this instead?
https://github.com/SatelliteQE/robottelo/blob/db19610b89dd4577b59f27555382c4aa736eaa3d/robottelo/host_helpers/api_factory.py#L144

I mean `baseos` and `appstream` are not custom repos, you could sync them as RH repos (just use `module_sca_manifest_org` instead of `module_org`) and ship them through the CV/LCE/AK used in your `content_for_client` fixture?

Reply:
I was just trying to get around the issue that I explain in the comment. I was trying to make as minimal changes as possible for the test since I'm just trying to convert to using newer rhel versions. I guess in this case I can revert the change and use a different helper like `enable_sync_redhat_repo`